lib/ruby_smb/smb2/pipe.rb

```diff
@@ -11,29 +11,45 @@ class Pipe < File
       #
       # @param peek_size [Integer] Amount of data to peek
       # @return [RubySMB::SMB2::Packet::IoctlResponse]
+      # @raise [RubySMB::Error::InvalidPacket] if not a valid FSCTL_PIPE_PEEK response
       def peek(peek_size: 0)
         packet = RubySMB::SMB2::Packet::IoctlRequest.new
         packet.ctl_code = RubySMB::Fscc::ControlCodes::FSCTL_PIPE_PEEK
         packet.flags.is_fsctl = true
         # read at least 16 bytes for state, avail, msg_count, first_msg_len
         packet.max_output_response = 16 + peek_size
         packet = set_header_fields(packet)
-        resp = @tree.client.send_recv(packet)
-        RubySMB::SMB2::Packet::IoctlResponse.read(resp)
+        raw_response = @tree.client.send_recv(packet)
+        begin
+          response = RubySMB::SMB2::Packet::IoctlResponse.read(raw_response)
+        rescue EOFError
```

Comment:
A begin/rescue EOFError like this happens in https://github.com/rapid7/ruby_smb/blob/master/lib/ruby_smb/generic_packet.rb#L45 returning either a RubySMB::SMB1::Packet::EmptyPacket or RubySMB::SMB1::Packet::ErrorPacket so you don't need to do that here.